explore/src/main/scala/explore/config/SpectroscopyModesTable.scala

```diff
@@ -218,35 +211,77 @@ object SpectroscopyModesTable {
   ): Boolean =
     rowToConf(row).exists(_ === conf)
 
-  protected val tableComponent =
-    ScalaFnComponent[ModesTableProps] { props =>
-      def toggleRow(row: SpectroscopyModeRow): Option[ScienceConfigurationData] =
-        rowToConf(row).filterNot(conf => props.scienceConfiguration.get.contains_(conf))
-
-      val tableInstance = ModesTableMaker.use(props.options)
-      <.div(
-        ExploreStyles.ModesTable,
-        ModesTableComponent(
-          table =
-            Table(celled = true, selectable = true, striped = true, compact = TableCompact.Very)(),
-          header = true,
-          headerCell = (c: ModesTableMaker.ColumnType) =>
-            TableHeaderCell(clazz = ExploreStyles.Sticky |+| ExploreStyles.ModesHeader)(
-              ^.textTransform.capitalize.when(c.id.toString =!= ResolutionColumnId.value),
-              ^.textTransform.none.when(c.id.toString === ResolutionColumnId.value)
-            ),
-          row = (rowData: Row[SpectroscopyModeRow]) =>
-            TableRow(
-              disabled = !enabledRow(rowData.original),
-              clazz = ExploreStyles.ModeSelected.when_(
-                props.scienceConfiguration.get.exists(conf => equalsConf(rowData.original, conf))
+  // protected def ensureContains(matrix: List[SpectroscopyModeRow], conf: Option[ScienceConfigurationData]): List[SpectroscopyModeRow] =
+  //   conf.fold(matrix)(c =>
+  //     if(matrix.exists(row => equalsConf(row, c)))
+  //       matrix
+  //     else
+
+  protected def enabledRow(row: SpectroscopyModeRow): Boolean =
+    List(Instrument.GmosNorth, Instrument.GmosSouth).contains_(row.instrument.instrument) &&
+      row.focalPlane === FocalPlane.SingleSlit
+
+  val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      .useMemoBy(_.spectroscopyRequirements)(props =>
```

Comment:
This looks super interesting, could you expand a bit on what it does?

Reply:
In this case, the filtered table (the 2nd parameter) will only be recomputed if the requirements (the 1st parameter) are different than in the last render. Difference in this case is according to `Reusability`. In other words, as long as `spectroscopyRequirements` can be reused, then the memoized list of filtered configurations is also reused instead of computing a fresh one.

The filtered table (whether it is a freshly computed one or a memoized one) is passed as an additional parameter to the `render` method.

Reply:
Coul we use this to get the spectroscpy matrix? Right now it is being loaded on every mount I think

Reply:
Maybe. The use case is a bit different since we want the fetch to happen asynchronously. I'll check.